ext/serializer.c

```diff
@@ -602,11 +602,11 @@ void ddtrace_set_root_span_properties(ddtrace_span_t *span) {
     zval *prop_name = ddtrace_spandata_property_name(span);
     if (strcmp(sapi_module.name, "cli") == 0) {
         ZVAL_STR(prop_type, zend_string_init(ZEND_STRL("cli"), 0));
-        if (SG(request_info).argc > 0) {
-            ZVAL_STR(prop_name, php_basename(SG(request_info).argv[0], strlen(SG(request_info).argv[0]), NULL, 0));
-        } else {
-            ZVAL_STR(prop_name, zend_string_init(ZEND_STRL("cli.command"), 0));
-        }
+        const char *script_name;
+        ZVAL_STR(prop_name,
+            (SG(request_info).argc > 0 && (script_name = SG(request_info).argv[0]) && script_name[0] != '\0')
```

Comment:
Just stating what we are doing
- check if we have arguments
- check if argument is not null
- check if we have at least one character
Then we use the script name, otherwise we use a literal "cli.command".
Sounds good !

Minor: Future you might like more comments

Reply:
For completeness, I have never been able to trigger `"cli.command"`. But, that's what the tracer is doing today, so that's what I'm doing in the profiler too. These additional checks I added would save us from some safety issues if it ever does happen, but also treats an empty string as the same as being not set, which is something we generally do for env vars.